document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/mapping/smallfull/mappers/ApiToPeriodMapperTest.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.company.account.CompanyAccountApi;
+import uk.gov.companieshouse.api.model.company.account.LastAccountsApi;
+import uk.gov.companieshouse.api.model.company.account.NextAccountsApi;
+import uk.gov.companieshouse.document.generator.accounts.mapping.smallfull.model.ixbrl.period.Period;
+
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ApiToPeriodMapperTest {
+
+    @Autowired
+    private ApiToPeriodMapper apiToPeriodMapper;
+
+    private static final String CURRENT_PERIOD_START_ON = "2018-01-01";
+
+    private static final String CURRENT_PERIOD_END_ON = "2018-12-31";
+
+    private static final String PREVIOUS_PERIOD_START_ON = "2017-01-01";
+
+    private static final String PREVIOUS_PERIOD_END_ON = "2017-12-31";
+
+    private static final String CURRENT_PERIOD_START_ON_FORMATTED = "1 January 2018";
+
+    private static final String CURRENT_PERIOD_END_ON_FORMATTED = "31 December 2018";
+
+    private static final String PREVIOUS_PERIOD_START_ON_FORMATTED = "1 January 2017";
+
+    private static final String PREVIOUS_PERIOD_END_ON_FORMATTED = "31 December 2017";
+
+    private static final String CURRENT_PERIOD_BS_DATE = "2018";
+
+    private static final String PREVIOUS_PERIOD_BS_DATE = "2017";
+
+    @Test
+    @DisplayName("tests that the dates from Api are mapped to Period IXBRL model for multi year filings")
+    public void testApiMapsDatesToPeriodModelForMultiYearFiling() {
+
+        Period period = ApiToPeriodMapper.INSTANCE.apiToPeriod(setAccountsFilingDates(true));
+
+        assertNotNull(period);
+        assertEquals(CURRENT_PERIOD_START_ON, period.getCurrentPeriodStartOn());
+        assertEquals(CURRENT_PERIOD_END_ON, period.getCurrentPeriodEndsOn());
+        assertEquals(PREVIOUS_PERIOD_START_ON, period.getPreviousPeriodStartOn());
+        assertEquals(PREVIOUS_PERIOD_END_ON, period.getPreviousPeriodEndsOn());
+        assertEquals(CURRENT_PERIOD_START_ON_FORMATTED, period.getCurrentPeriodStartOnFormatted());
+        assertEquals(CURRENT_PERIOD_END_ON_FORMATTED, period.getCurrentPeriodEndOnFormatted());
+        assertEquals(PREVIOUS_PERIOD_START_ON_FORMATTED, period.getPreviousPeriodStartOnFormatted());
+        assertEquals(PREVIOUS_PERIOD_END_ON_FORMATTED, period.getPreviousPeriodEndOnFormatted());
+        assertEquals(CURRENT_PERIOD_BS_DATE, period.getCurrentPeriodBSDate());
+        assertEquals(PREVIOUS_PERIOD_BS_DATE, period.getPreviousPeriodBSDate());
+    }
+
+    @Test
+    @DisplayName("tests that the dates from Api are mapped to Period IXBRL model for single year filings")
+    public void testApiMapsDatesToPeriodModelForSingleYearFiling() {
```

Comment:
I would remove the `public` modifier as it is not needed in Junit 5

Reply:
done